diff --git a/src/podofo/main/PdfFont.cpp b/src/podofo/main/PdfFont.cpp
--- a/src/podofo/main/PdfFont.cpp
+++ b/src/podofo/main/PdfFont.cpp
@@ -6,4 +6,5 @@
 #include "PdfFont.h"
 
 #include <utf8cpp/utf8.h>
+#include <algorithm>
 
@@ -1112,6 +1113,10 @@ vector<PdfCharGIDInfo> PdfFont::GetCharGIDInfos() const
                 ret[i] = { *pair.second, *pair.second, PdfGID(*pair.second)};
                 i++;
             }
+            std::sort(ret.begin(), ret.end(), [](const PdfCharGIDInfo& lhs, const PdfCharGIDInfo& rhs) {
+                return lhs.Cid < rhs.Cid;
+            });
             return ret;
         }
 
@@ -1147,5 +1152,8 @@ vector<PdfCharGIDInfo> PdfFont::GetCharGIDInfos() const
         }
     }
 
+    std::sort(ret.begin(), ret.end(), [](const PdfCharGIDInfo& lhs, const PdfCharGIDInfo& rhs) {
+        return lhs.Cid < rhs.Cid;
+    });
     return ret;
 }
