diff --git a/MPC/config/zlib.mpb b/MPC/config/zlib.mpb
index 731935a..58ca233 100644
--- a/MPC/config/zlib.mpb
+++ b/MPC/config/zlib.mpb
@@ -14,9 +14,11 @@ feature(zlib) {
   macros   += ZLIB
 
   specific(prop:windows) {
-    Debug::lit_libs += zlibd
-    Release::lit_libs += zlib
+    zliblib = zlib
+    Debug::zliblib = zlibd
   } else {
-    lit_libs += z
+    zliblib = z
   }
+
+  lit_libs += $(ZLIBLIB)
 }
