diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3c4aaf4..c154a9f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -236,7 +236,7 @@ if(MSVC)
   list(APPEND CMAKE_REQUIRED_LINK_OPTIONS -link setargv.obj)
   list(APPEND WINDOWS_WMAIN_LINK -link wsetargv.obj)
 endif()
-if(MINGW)
+if(MINGW AND NOT DEFINED ENV{QPDF_BCC64X})
   get_filename_component(MINGW_BIN_DIR "${CMAKE_CXX_COMPILER}" DIRECTORY)
   execute_process(
     COMMAND gcc --print-file-name=CRT_glob.o
diff --git a/qpdf/CMakeLists.txt b/qpdf/CMakeLists.txt
index 877d8af..3f8c6b1 100644
--- a/qpdf/CMakeLists.txt
+++ b/qpdf/CMakeLists.txt
@@ -73,7 +73,7 @@ install(TARGETS qpdf fix-qdf
   TYPE RUNTIME
   COMPONENT ${COMPONENT_CLI})
 
-if(MINGW)
+if(MINGW AND NOT DEFINED ENV{QPDF_BCC64X})
   # For MSVC, including InstallRequiredSystemLibraries in the
   # top-level CMakeLists.txt is sufficient. For mingw, we have to copy
   # mingw libraries in ourselves.
