diff --git a/libarchive/archive_endian.h b/libarchive/archive_endian.h
index 939acc6..f78b1e1 100644
--- a/libarchive/archive_endian.h
+++ b/libarchive/archive_endian.h
@@ -44,7 +44,7 @@
  * - IBM VisualAge 6 (XL v6)
  * - Sun WorkShop C (SunPro) before 5.9
  */
-#if defined(__WATCOMC__) || defined(__sgi) || defined(__hpux) || defined(__BORLANDC__)
+#if defined(__WATCOMC__) || defined(__sgi) || defined(__hpux) || (defined(__BORLANDC__) && !defined(__clang__))
 #define	inline
 #elif defined(__IBMC__) && __IBMC__ < 700
 #define	inline
diff --git a/libarchive/archive_integer.h b/libarchive/archive_integer.h
index 1e71592..85487cc 100644
--- a/libarchive/archive_integer.h
+++ b/libarchive/archive_integer.h
@@ -78,7 +78,7 @@
  * - IBM VisualAge 6 (XL v6)
  * - Sun WorkShop C (SunPro) before 5.9
  */
-#if defined(__WATCOMC__) || defined(__sgi) || defined(__hpux) || defined(__BORLANDC__)
+#if defined(__WATCOMC__) || defined(__sgi) || defined(__hpux) || (defined(__BORLANDC__) && !defined(__clang__))
 #define	inline
 #elif defined(__IBMC__) && __IBMC__ < 700
 #define	inline
diff --git a/libarchive/archive_platform.h b/libarchive/archive_platform.h
index 33dc558..41a1860 100644
--- a/libarchive/archive_platform.h
+++ b/libarchive/archive_platform.h
@@ -99,7 +99,7 @@
 #endif
 
 /* Borland warns about its own constants!  */
-#if defined(__BORLANDC__)
+#if defined(__BORLANDC__) && !defined(__clang__)
 # if HAVE_DECL_UINT64_MAX
 #  undef	UINT64_MAX
 #  undef	HAVE_DECL_UINT64_MAX
diff --git a/libarchive/archive_ppmd_private.h b/libarchive/archive_ppmd_private.h
index bc525b4..36d0b40 100644
--- a/libarchive/archive_ppmd_private.h
+++ b/libarchive/archive_ppmd_private.h
@@ -39,7 +39,7 @@ typedef unsigned long UInt64;
 
 #else
 
-#if defined(_MSC_VER) || defined(__BORLANDC__)
+#if defined(_MSC_VER) || (defined(__BORLANDC__) && !defined(__clang__))
 typedef __int64 Int64;
 typedef unsigned __int64 UInt64;
 #define UINT64_CONST(n) n
diff --git a/libarchive/archive_private.h b/libarchive/archive_private.h
index 0f06863..512793f 100644
--- a/libarchive/archive_private.h
+++ b/libarchive/archive_private.h
@@ -198,7 +198,7 @@ void __archive_reset_read_data(struct archive *);
 
 #define	err_combine(a,b)	((a) < (b) ? (a) : (b))
 
-#if defined(__BORLANDC__) || (defined(_MSC_VER) &&  _MSC_VER <= 1300)
+#if (defined(__BORLANDC__) && !defined(__clang__)) || (defined(_MSC_VER) &&  _MSC_VER <= 1300)
 # define	ARCHIVE_LITERAL_LL(x)	x##i64
 # define	ARCHIVE_LITERAL_ULL(x)	x##ui64
 #else
diff --git a/libarchive/archive_windows.c b/libarchive/archive_windows.c
index b787a0d..9181bb3 100644
--- a/libarchive/archive_windows.c
+++ b/libarchive/archive_windows.c
@@ -357,7 +357,7 @@ __la_open(const char *path, int flags, ...)
 		}
 	}
 	if (ws == NULL) {
-#if defined(__BORLANDC__)
+#if defined(__BORLANDC__) && !defined(__clang__)
 		/* Borland has no mode argument.
 		   TODO: Fix mode of new file.  */
 		r = _open(path, flags);
diff --git a/libarchive/archive_windows.h b/libarchive/archive_windows.h
index fef7cf9..9af64ae 100644
--- a/libarchive/archive_windows.h
+++ b/libarchive/archive_windows.h
@@ -74,7 +74,7 @@
 
 #include "archive_platform_stat.h"
 
-#if defined(__BORLANDC__)
+#if defined(__BORLANDC__) && !defined(__clang__)
 #pragma warn -8068	/* Constant out of range in comparison. */
 #pragma warn -8072	/* Suspicious pointer arithmetic. */
 #endif
@@ -97,7 +97,7 @@
 #undef fstat
 #endif
 #define	fstat		__la_fstat
-#if !defined(__BORLANDC__)
+#if !defined(__BORLANDC__) || defined(__clang__)
 #ifdef lseek
 #undef lseek
 #endif
@@ -294,7 +294,7 @@ extern void __la_dosmaperr(unsigned long e);
 extern struct archive_entry *__la_win_entry_in_posix_pathseparator(
     struct archive_entry *);
 
-#if defined(HAVE_WCRTOMB) && defined(__BORLANDC__)
+#if defined(HAVE_WCRTOMB) && defined(__BORLANDC__) && !defined(__clang__)
 typedef int mbstate_t;
 size_t wcrtomb(char *, wchar_t, mbstate_t *);
 #endif
diff --git a/libarchive/filter_fork_windows.c b/libarchive/filter_fork_windows.c
index a0fd638..9fbf1c2 100644
--- a/libarchive/filter_fork_windows.c
+++ b/libarchive/filter_fork_windows.c
@@ -77,7 +77,7 @@ load_WaitForInputIdle(PINIT_ONCE InitOnce, PVOID Parameter, PVOID *Context) {
 	(void)Parameter; /* UNUSED */
 
 	*Context = (user32 != NULL) ?
-	    (PVOID)GetProcAddress(user32, "WaitForInputIdle") : failing_wait;
+	    (PVOID)GetProcAddress(user32, "WaitForInputIdle") : (PVOID)failing_wait;
 
 	return TRUE;
 }
diff --git a/libarchive/test/test_open_fd.c b/libarchive/test/test_open_fd.c
index bd73fb3..fecbcbb 100644
--- a/libarchive/test/test_open_fd.c
+++ b/libarchive/test/test_open_fd.c
@@ -43,7 +43,7 @@ DEFINE_TEST(test_open_fd)
 	int fd;
 	const char *skip_open_fd_err_test;
 
-#if defined(__BORLANDC__)
+#if defined(__BORLANDC__) && !defined(__clang__)
 	fd = open("test.tar", O_RDWR | O_CREAT | O_BINARY);
 #else
 	fd = open("test.tar", O_RDWR | O_CREAT | O_BINARY, 0777);
diff --git a/libarchive/test/test_read_data_large.c b/libarchive/test/test_read_data_large.c
index 3718234..bb455c8 100644
--- a/libarchive/test/test_read_data_large.c
+++ b/libarchive/test/test_read_data_large.c
@@ -35,6 +35,9 @@
 #if defined(_WIN32) && !defined(__CYGWIN__)
 #define open _open
 #define close _close
+#ifdef ftruncate
+#undef ftruncate
+#endif
 #define ftruncate _chsize
 #endif
 
@@ -92,7 +95,7 @@ DEFINE_TEST(test_read_data_large)
 	assertA(0 == archive_read_support_filter_all(a));
 	assertA(0 == archive_read_open_memory(a, buff1, sizeof(buff1)));
 	assertA(0 == archive_read_next_header(a, &ae));
-#if defined(__BORLANDC__)
+#if defined(__BORLANDC__) && !defined(__clang__)
 	tmpfilefd = open(tmpfilename, O_WRONLY | O_CREAT | O_BINARY);
 #else
 	tmpfilefd = open(tmpfilename, O_WRONLY | O_CREAT | O_BINARY, 0777);
diff --git a/libarchive/test/test_read_format_gtar_sparse_skip_entry.c b/libarchive/test/test_read_format_gtar_sparse_skip_entry.c
index eaf015b..40b08b3 100644
--- a/libarchive/test/test_read_format_gtar_sparse_skip_entry.c
+++ b/libarchive/test/test_read_format_gtar_sparse_skip_entry.c
@@ -24,7 +24,7 @@
  */
 #include "test.h"
 
-#if defined(__BORLANDC__) || (defined(_MSC_VER) &&  _MSC_VER <= 1300)
+#if (defined(__BORLANDC__) && !defined(__clang__)) || (defined(_MSC_VER) &&  _MSC_VER <= 1300)
 # define	LITERAL_LL(x)	x##i64
 #else
 # define	LITERAL_LL(x)	x##ll
diff --git a/libarchive/test/test_read_large.c b/libarchive/test/test_read_large.c
index 46dee4e..784b6ff 100644
--- a/libarchive/test/test_read_large.c
+++ b/libarchive/test/test_read_large.c
@@ -73,7 +73,7 @@ DEFINE_TEST(test_read_large)
 	assertA(0 == archive_read_support_filter_all(a));
 	assertA(0 == archive_read_open_memory(a, buff, sizeof(buff)));
 	assertA(0 == archive_read_next_header(a, &entry));
-#if defined(__BORLANDC__)
+#if defined(__BORLANDC__) && !defined(__clang__)
 	tmpfilefd = open(tmpfilename, O_WRONLY | O_CREAT | O_BINARY);
 #else
 	tmpfilefd = open(tmpfilename, O_WRONLY | O_CREAT | O_BINARY, 0755);
diff --git a/tar/write.c b/tar/write.c
index 13ff4ea..f9b661a 100644
--- a/tar/write.c
+++ b/tar/write.c
@@ -276,7 +276,7 @@ tar_mode_r(struct bsdtar *bsdtar)
 
 	format = ARCHIVE_FORMAT_TAR_PAX_RESTRICTED;
 
-#if defined(__BORLANDC__)
+#if defined(__BORLANDC__) && !defined(__clang__)
 	bsdtar->fd = open(bsdtar->filename, O_RDWR | O_CREAT | O_BINARY);
 #else
 	bsdtar->fd = open(bsdtar->filename, O_RDWR | O_CREAT | O_BINARY, 0666);
diff --git a/test_utils/test_common.h b/test_utils/test_common.h
index 2f19597..fed244c 100644
--- a/test_utils/test_common.h
+++ b/test_utils/test_common.h
@@ -112,7 +112,7 @@
 #define snprintf	sprintf_s
 #endif
 
-#if defined(__BORLANDC__)
+#if defined(__BORLANDC__) && !defined(__clang__)
 #pragma warn -8068	/* Constant out of range in comparison. */
 #endif
 
