diff --git a/src/xercesc/sax/HandlerBase.hpp b/src/xercesc/sax/HandlerBase.hpp
--- a/src/xercesc/sax/HandlerBase.hpp
+++ b/src/xercesc/sax/HandlerBase.hpp
@@ -357,7 +357,11 @@ public:
     //@}
 
     HandlerBase() {};
+#if defined(__BORLANDC__) && defined(__clang__)
+    virtual ~HandlerBase();
+#else
     virtual ~HandlerBase() {};
+#endif
 
 private:
 	// -----------------------------------------------------------------------
diff --git a/src/xercesc/sax/Dummy.cpp b/src/xercesc/sax/Dummy.cpp
--- a/src/xercesc/sax/Dummy.cpp
+++ b/src/xercesc/sax/Dummy.cpp
@@ -32,4 +32,7 @@
 #include <xercesc/sax/SAXParseException.hpp>
 
 XERCES_CPP_NAMESPACE_BEGIN
+#if defined(__BORLANDC__) && defined(__clang__)
+HandlerBase::~HandlerBase() = default;
+#endif
 XERCES_CPP_NAMESPACE_END
