diff --git a/third_party/third_party.gni b/third_party/third_party.gni
index 1fe96e9..3c6feed 100644
--- a/third_party/third_party.gni
+++ b/third_party/third_party.gni
@@ -107,7 +107,17 @@ template("system") {
   config(target_name + "_public") {
     forward_variables_from(invoker, "*", [])
   }
+
+  if (skia_use_bcc64x && defined(invoker.ldflags)) {
+    config(target_name + "_bcc64x_link") {
+      ldflags = invoker.ldflags
+    }
+  }
+
   group(target_name) {
     public_configs = [ ":" + target_name + "_public" ]
+    if (skia_use_bcc64x && defined(invoker.ldflags)) {
+      all_dependent_configs = [ ":" + target_name + "_bcc64x_link" ]
+    }
   }
 }

diff --git a/gn/skia/BUILD.gn b/gn/skia/BUILD.gn
--- a/gn/skia/BUILD.gn
+++ b/gn/skia/BUILD.gn
@@ -119,8 +119,10 @@ config("default") {
   } else {
-    cflags += [
-      "-fPIC",
-      "-fvisibility=hidden",
-    ]
-    cflags_cc += [ "-fvisibility-inlines-hidden" ]
+    cflags += [ "-fPIC" ]
+    if (is_bcc64x_toolchain) {
+      ldflags += [ "-Wl,--export-all-symbols" ]
+    } else {
+      cflags += [ "-fvisibility=hidden" ]
+      cflags_cc += [ "-fvisibility-inlines-hidden" ]
+    }
     cflags_cc += [ "-std=c++20" ]
   }
