diff --git a/third_party/freetype2/BUILD.gn b/third_party/freetype2/BUILD.gn
index 631b909..5ca8178 100644
--- a/third_party/freetype2/BUILD.gn
+++ b/third_party/freetype2/BUILD.gn
@@ -25,7 +25,12 @@ import("../third_party.gni")
 if (skia_use_system_freetype2) {
   system("freetype2") {
     include_dirs = [ skia_system_freetype2_include_path ]
-    libs = [ skia_system_freetype2_lib ]
+    if (skia_use_bcc64x) {
+      ldflags = [ skia_system_freetype2_lib ]
+      defines = [ "SK_FREETYPE_MINIMUM_RUNTIME_VERSION_IS_BUILD_VERSION=1" ]
+    } else {
+      libs = [ skia_system_freetype2_lib ]
+    }
   }
 } else {
   third_party("freetype2") {
