diff --git a/gn/BUILDCONFIG.gn b/gn/BUILDCONFIG.gn
--- a/gn/BUILDCONFIG.gn
+++ b/gn/BUILDCONFIG.gn
@@ -183,6 +183,10 @@ template("component") {
   }
 }
 
+declare_args() {
+  skia_use_bcc64x = false
+}
+
 # Default configs
 default_configs = [
   "//gn/skia:default",
@@ -248,7 +252,13 @@ skia_header_target_default_configs = []
       [ "//gn/skia:warnings_for_public_headers" ]
 }
 
-if (is_win) {
+if (is_win && skia_use_bcc64x) {
+  # BCC64X is the Windows target compiler.  Skia host tools remain on the
+  # native Windows host toolchain.
+  set_default_toolchain("//gn/toolchain:gcc_like")
+  default_toolchain_name = "gcc_like"
+  host_toolchain = "msvc_host"
+} else if (is_win) {
   # Windows tool chain
   set_default_toolchain("//gn/toolchain:msvc")
   default_toolchain_name = "msvc"
diff --git a/gn/skia/BUILD.gn b/gn/skia/BUILD.gn
--- a/gn/skia/BUILD.gn
+++ b/gn/skia/BUILD.gn
@@ -8,6 +8,8 @@ if (is_wasm) {
 
 import("../skia.gni")
 
+is_bcc64x_toolchain = skia_use_bcc64x && current_toolchain == default_toolchain
+
 declare_args() {
   extra_asmflags = []
   extra_cflags = []
@@ -52,7 +54,7 @@ config("default") {
   # (These unknown attribute warnings are on by default, so we don't make
   # disabling them part of :warnings, as some targets remove :warnings.)
-  if (is_win && !is_clang) {
+  if (is_win && !is_clang && !is_bcc64x_toolchain) {
     cflags += [
       "/wd5030",
       "/wd5051",
@@ -66,7 +68,7 @@ config("default") {
     cflags += [ "-ffp-contract=off" ]
   }
 
-  if (is_win) {
+  if (is_win && !is_bcc64x_toolchain) {
     if (is_clang && current_cpu == "arm64") {
       cflags += [ "--target=arm64-windows" ]
     }
@@ -406,7 +408,7 @@ config("recover_pointer_overflow") {
 
 config("no_exceptions") {
   # Exceptions are disabled by default on Windows.  (Use /EHsc to enable them.)
-  if (!is_win) {
+  if (!is_win || is_bcc64x_toolchain) {
     cflags_cc = [ "-fno-exceptions" ]
     cflags_objcc = cflags_cc
   }
@@ -418,14 +420,14 @@ config("warnings") {
   cflags_objcc = []
 
   if (werror) {
-    if (is_win) {
+    if (is_win && !is_bcc64x_toolchain) {
       cflags += [ "/WX" ]
     } else {
       cflags += [ "-Werror" ]
     }
   }
 
-  if (is_win) {
+  if (is_win && !is_bcc64x_toolchain) {
     cflags += [
       "/W3",  # Turn on lots of warnings.
 
@@ -676,7 +678,9 @@ config("debug_symbols") {
       "-gline-tables-only",
       "-funwind-tables",  # Helps make in-process backtraces fuller.
     ]
-  } else if (is_win) {
+  } else if (is_win && is_bcc64x_toolchain) {
+    cflags = [ "-g" ]
+  } else if (is_win) {
     cflags = [ "/Z7" ]
     if (is_clang) {
       cflags += [ "-gcodeview-ghash" ]
@@ -706,7 +710,7 @@ config("no_rtti") {
   # -fsanitize=undefined
   if (sanitize != "ASAN") {
-    if (is_win) {
+    if (is_win && !is_bcc64x_toolchain) {
       cflags_cc = [ "/GR-" ]
     } else {
       cflags_cc = [ "-fno-rtti" ]
@@ -717,7 +721,7 @@ config("no_rtti") {
 
 config("optimize") {
   ldflags = []
-  if (is_win) {
+  if (is_win && !is_bcc64x_toolchain) {
     cflags = [
       "/O2",
       "/Zc:inline",
diff --git a/BUILD.gn b/BUILD.gn
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -130,7 +130,7 @@ opts("ml3") {
   enabled = is_x86
   sources = skia_opts.ml3_sources
-  if (is_win) {
+  if (is_win && !skia_use_bcc64x) {
     cflags = [ "/arch:AVX2" ]
   } else if (current_cpu == "x86") {
     cflags = [ "-march=haswell" ]
@@ -143,7 +143,7 @@ opts("ml4") {
   enabled = is_x86
   sources = skia_opts.ml4_sources
-  if (is_win) {
+  if (is_win && !skia_use_bcc64x) {
     cflags = [ "/arch:AVX512" ]
   } else if (current_cpu == "x86") {
     cflags = [ "-march=skylake-avx512" ]
diff --git a/modules/skcms/BUILD.gn b/modules/skcms/BUILD.gn
--- a/modules/skcms/BUILD.gn
+++ b/modules/skcms/BUILD.gn
@@ -31,7 +31,13 @@ arch("skcms_TransformHsw") {
   enabled = current_cpu == "x64" && target_os != "android"
   sources = skcms_TransformHsw
   if (is_win) {
-    if (is_clang) {
+    if (skia_use_bcc64x) {
+      cflags = [
+        "-mavx2",
+        "-mf16c",
+        "-ffp-contract=off",
+      ]
+    } else if (is_clang) {
       cflags = [
         "/clang:-mavx2",
         "/clang:-mf16c",
@@ -57,7 +63,16 @@ arch("skcms_TransformSkx") {
   enabled = current_cpu == "x64" && target_os != "android"
   sources = skcms_TransformSkx
   if (is_win) {
-    if (is_clang) {
+    if (skia_use_bcc64x) {
+      cflags = [
+        "-mavx512f",
+        "-mavx512dq",
+        "-mavx512cd",
+        "-mavx512bw",
+        "-mavx512vl",
+        "-ffp-contract=off",
+      ]
+    } else if (is_clang) {
       cflags = [
         "/clang:-mavx512f",
         "/clang:-mavx512dq",
diff --git a/third_party/third_party.gni b/third_party/third_party.gni
--- a/third_party/third_party.gni
+++ b/third_party/third_party.gni
@@ -83,7 +83,9 @@ template("third_party") {
       if (!defined(cflags)) {
         cflags = []
       }
-      if (is_win) {
+      if (is_win && skia_use_bcc64x && current_toolchain == default_toolchain) {
+        cflags += [ "-w" ]
+      } else if (is_win) {
         cflags += [ "/w" ]
       } else {
         cflags += [ "-w" ]
diff --git a/gn/toolchain/BUILD.gn b/gn/toolchain/BUILD.gn
--- a/gn/toolchain/BUILD.gn
+++ b/gn/toolchain/BUILD.gn
@@ -6,6 +6,7 @@ declare_args() {
   host_ar = ar
   host_cc = cc
   host_cxx = cxx
+  python_executable = "python3"
 
   if (is_android) {
     _prefix = "$ndk/toolchains/llvm/prebuilt/$ndk_host/bin"
@@ -211,7 +212,7 @@ template("msvc_toolchain") {
 
     tool("copy") {
       cp_py = rebase_path("../cp.py")
-      command = "$shell python3 \"$cp_py\" {{source}} {{output}}"
+      command = "$shell $python_executable \"$cp_py\" {{source}} {{output}}"
       description = "copy {{source}} {{output}}"
     }
@@ -246,6 +247,10 @@ template("gcc_like_toolchain") {
     link = invoker.link
     lib_switch = "-l"
     lib_dir_switch = "-L"
+    if (current_os == "win" && skia_use_bcc64x &&
+        current_toolchain == default_toolchain) {
+      lib_switch = ""
+    }
 
     current_cpu = invoker.cpu
     current_os = invoker.os
@@ -315,7 +320,7 @@ template("gcc_like_toolchain") {
         rspfile = "{{output}}.rsp"
         rspfile_content = "{{inputs}}"
         rm_py = rebase_path("../rm.py")
-        command = "$shell python3 \"$rm_py\" \"{{output}}\" && $ar rcs {{output}} @$rspfile"
+        command = "$shell $python_executable \"$rm_py\" \"{{output}}\" && $ar rcs {{output}} @$rspfile"
       }
 
       outputs =
@@ -335,7 +340,29 @@ template("gcc_like_toolchain") {
     }
 
     tool("solink") {
-      soname = "{{target_output_name}}{{output_extension}}"
+      if (current_os == "win" && skia_use_bcc64x &&
+          current_toolchain == default_toolchain) {
+        dllname = "{{root_out_dir}}/{{target_output_name}}{{output_extension}}"
+        libname = "{{root_out_dir}}/{{target_output_name}}.lib"
+        rspfile = "${dllname}.rsp"
+        command = "$shell $link --rsp-quoting=windows -L\"%CB_BDS%/lib/win64x/release\" -L\"%CB_BDS%/lib/win64x/release/psdk\" -tD -o$dllname @$rspfile -Wl,--out-implib,$libname"
+        rspfile_content = "{{inputs}} {{libs}} {{solibs}} {{ldflags}}"
+        outputs = [
+          dllname,
+          libname,
+        ]
+        default_output_extension = ".dll"
+        default_output_dir = "{{root_out_dir}}"
+        link_output = libname
+        depend_output = libname
+        runtime_outputs = [ dllname ]
+        restat = true
+        description = "link {{output}}"
+        if (0 <= link_pool_depth) {
+          pool = ":link_pool($default_toolchain)"
+        }
+      } else {
+        soname = "{{target_output_name}}{{output_extension}}"
 
-      rpath = "-Wl,-soname,$soname"
-      if (current_os == "mac" || current_os == "ios" || current_os == "tvos") {
-        rpath = "-Wl,-install_name,@rpath/$soname"
-      }
+        rpath = "-Wl,-soname,$soname"
+        if (current_os == "mac" || current_os == "ios" || current_os == "tvos") {
+          rpath = "-Wl,-install_name,@rpath/$soname"
+        }
 
-      rspfile = "{{output}}.rsp"
-      rspfile_content = "{{inputs}}"
+        rspfile = "{{output}}.rsp"
+        rspfile_content = "{{inputs}}"
@@ -363,22 +390,23 @@ template("gcc_like_toolchain") {
-      command = "$link -shared {{ldflags}} $_start_group @$rspfile {{frameworks}} {{solibs}} $_end_group {{libs}} $rpath $_emcc_flags -o {{output}}"
-      outputs = [ "{{root_out_dir}}/$soname" ]
-      output_prefix = "lib"
-      if (current_os == "mac" || current_os == "ios" || current_os == "tvos") {
-        default_output_extension = ".dylib"
-      } else if (current_os == "wasm" && !is_canvaskit) {
-        default_output_extension = ".wasm.so"
-      } else {
-        default_output_extension = ".so"
-      }
-      description = "link {{output}}"
-      if (0 <= link_pool_depth) {
-        pool = ":link_pool($default_toolchain)"
+        command = "$link -shared {{ldflags}} $_start_group @$rspfile {{frameworks}} {{solibs}} $_end_group {{libs}} $rpath $_emcc_flags -o {{output}}"
+        outputs = [ "{{root_out_dir}}/$soname" ]
+        output_prefix = "lib"
+        if (current_os == "mac" || current_os == "ios" || current_os == "tvos") {
+          default_output_extension = ".dylib"
+        } else if (current_os == "wasm" && !is_canvaskit) {
+          default_output_extension = ".wasm.so"
+        } else {
+          default_output_extension = ".so"
+        }
+        description = "link {{output}}"
+        if (0 <= link_pool_depth) {
+          pool = ":link_pool($default_toolchain)"
+        }
       }
     }
@@ -427,13 +455,13 @@ template("gcc_like_toolchain") {
 
     tool("copy") {
       cp_py = rebase_path("../cp.py")
-      command = "python3 \"$cp_py\" {{source}} {{output}}"
+      command = "$python_executable \"$cp_py\" {{source}} {{output}}"
       description = "copy {{source}} {{output}}"
     }
 
     tool("copy_bundle_data") {
       cp_py = rebase_path("../cp.py")
-      command = "python3 \"$cp_py\" {{source}} {{output}}"
+      command = "$python_executable \"$cp_py\" {{source}} {{output}}"
       description = "copy_bundle_data {{source}} {{output}}"
     }
