diff --git a/gn/toolchain/BUILD.gn b/gn/toolchain/BUILD.gn
--- a/gn/toolchain/BUILD.gn
+++ b/gn/toolchain/BUILD.gn
@@ -7,6 +7,7 @@ declare_args() {
   host_cc = cc
   host_cxx = cxx
   python_executable = "python3"
   bcc64x_rad_lib_dir = ""
   bcc64x_rad_psdk_lib_dir = ""
+  bcc64x_ucrt_compat_library = ""
 
@@ -338,8 +339,9 @@ template("gcc_like_toolchain") {
       if (current_os == "win" && skia_use_bcc64x &&
           current_toolchain == default_toolchain) {
         dllname = "{{root_out_dir}}/{{target_output_name}}{{output_extension}}"
         libname = "{{root_out_dir}}/{{target_output_name}}.lib"
         rspfile = "${dllname}.rsp"
-        command = "$shell $link --rsp-quoting=windows -L\"$bcc64x_rad_lib_dir\" -L\"$bcc64x_rad_psdk_lib_dir\" -tD -o$dllname @$rspfile -Wl,--out-implib,$libname"
+        assert(bcc64x_ucrt_compat_library != "", "BCC64X UCRT compatibility archive is required")
+        command = "$shell $link --rsp-quoting=windows -L\"$bcc64x_rad_lib_dir\" -L\"$bcc64x_rad_psdk_lib_dir\" -tD -o$dllname @$rspfile \"$bcc64x_ucrt_compat_library\" -Wl,--out-implib,$libname"
         rspfile_content = "{{inputs}} {{libs}} {{solibs}} {{ldflags}}"
         outputs = [
