diff --git a/src/gpu/ganesh/d3d/GrD3DGpu.cpp b/src/gpu/ganesh/d3d/GrD3DGpu.cpp
--- a/src/gpu/ganesh/d3d/GrD3DGpu.cpp
+++ b/src/gpu/ganesh/d3d/GrD3DGpu.cpp
@@ -1637,7 +1637,7 @@ bool GrD3DGpu::isTestingOnlyBackendTexture(const GrBackendTexture& tex) const {
     if (!textureResource) {
         return false;
     }
-    return !(textureResource->GetDesc().Flags & D3D12_RESOURCE_FLAG_DENY_SHADER_RESOURCE);
+    return !(GrD3DGetResourceDesc(textureResource).Flags & D3D12_RESOURCE_FLAG_DENY_SHADER_RESOURCE);
 }
 
 GrBackendRenderTarget GrD3DGpu::createTestingOnlyBackendRenderTarget(SkISize dimensions,
