--- a/src/pqxx-source.hxx
+++ b/src/pqxx-source.hxx
@@ -27,7 +27,7 @@
 // We're building libpqxx as a shared library.
 #    undef PQXX_LIBEXPORT
 #    define PQXX_LIBEXPORT __declspec(dllexport)
-#    define PQXX_PRIVATE __declspec()
+#    define PQXX_PRIVATE
 #  endif // PQXX_SHARED
 
 #endif // _WIN32
--- a/include/pqxx/internal/header-pre.hxx
+++ b/include/pqxx/internal/header-pre.hxx
@@ -73,7 +73,9 @@
 #endif
 
 
-#if __has_cpp_attribute(gnu::cold)
+#if defined(__clang__) && defined(__BORLANDC__)
+#  define PQXX_COLD __attribute__((cold))
+#elif __has_cpp_attribute(gnu::cold)
 /// Tell the compiler to optimise a function for size, not speed.
 #  define PQXX_COLD [[gnu::cold]]
 #else
--- a/include/pqxx/internal/header-pre.hxx
+++ b/include/pqxx/internal/header-pre.hxx
@@ -65,7 +65,9 @@
 #endif
 
 
-#if __has_cpp_attribute(gnu::hot)
+#if defined(__clang__) && defined(__BORLANDC__)
+#  define PQXX_HOT __attribute__((hot))
+#elif __has_cpp_attribute(gnu::hot)
 /// Tell the compiler that this function can get called a lot.
 #  define PQXX_HOT [[gnu::hot]]
 #else
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -48,9 +48,17 @@
 endif()
 # cmake-lint: disable=C0301
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_SIMULATE_ID STREQUAL "EMBT")
+    set(PQXX_OUTPUT_NAME pqxx)
+else()
+    set(PQXX_OUTPUT_NAME $<IF:$<PLATFORM_ID:Windows>,pqxx,pqxx-${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}>)
+endif()
 set_target_properties(
     pqxx PROPERTIES
-    OUTPUT_NAME $<IF:$<PLATFORM_ID:Windows>,pqxx,pqxx-${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}>
+    PREFIX ""
+    IMPORT_PREFIX ""
+    OUTPUT_NAME ${PQXX_OUTPUT_NAME}
 )
+unset(PQXX_OUTPUT_NAME)
 LIBRARY_TARGET_SETUP(pqxx)
 
 # Install pkg-config file.
