diff --git a/include/pqxx/internal/header-pre.hxx b/include/pqxx/internal/header-pre.hxx
--- a/include/pqxx/internal/header-pre.hxx
+++ b/include/pqxx/internal/header-pre.hxx
@@ -65,7 +65,9 @@
 #endif
 
 
-#if __has_cpp_attribute(gnu::hot)
+#if defined(__clang__) && defined(__BORLANDC__)
+#  define PQXX_HOT __attribute__((hot))
+#elif __has_cpp_attribute(gnu::hot)
 /// Tell the compiler that this function can get called a lot.
 #  define PQXX_HOT [[gnu::hot]]
 #else
@@ -73,7 +75,9 @@
 #endif
 
 
-#if __has_cpp_attribute(gnu::cold)
+#if defined(__clang__) && defined(__BORLANDC__)
+#  define PQXX_COLD __attribute__((cold))
+#elif __has_cpp_attribute(gnu::cold)
 /// Tell the compiler to optimise a function for size, not speed.
 #  define PQXX_COLD [[gnu::cold]]
 #else
